subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_iso.c

```diff
@@ -627,9 +627,24 @@ static void isr_tx_common(void *param,
 	if (!pdu) {
 		uint8_t payload_index;
 
-		payload_index = (lll->bn_curr - 1U) +
-				(lll->ptc_curr * lll->pto);
-		payload_count = lll->payload_count + payload_index - lll->bn;
+		if (lll->ptc_curr) {
+			uint8_t ptx_idx = lll->ptc_curr - 1;
+			uint8_t ptx_payload_idx;
+			uint32_t ptx_group_mult;
+			uint8_t ptx_group_idx;
+
+			/* Calculate group index and multiplier for deriving
+			 * pre-transmission payload index.
+			 */
+			ptx_group_idx = ptx_idx / lll->bn;
+			ptx_payload_idx = ptx_idx - ptx_group_idx * lll->bn;
+			ptx_group_mult = (ptx_group_idx + 1) * lll->pto;
+			payload_index = ptx_payload_idx + ptx_group_mult * lll->bn;
+		} else {
+			payload_index  = lll->bn_curr - 1U;
+		}
+
+		payload_count = lll->payload_count - lll->bn + payload_index;
```

Comment:
Similarly can `lll->bn` be larger than `lll->payload_count` here?

Reply:
No, `lll->payload_count` was pre-incremented by `lll->bn`  here: https://github.com/zephyrproject-rtos/zephyr/blob/42faa9438f8b0ddb827718b3d20699f9bf88e0c0/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_iso.c#L206-L207

Reply:
If `lll->latency_prepare` then that comment doesn't hold up. Can `lll->latency_prepare == 0`?

Reply:
`lll->latency_prepare` is greater than `1` when the `lll->payload_count` is incremented.

https://github.com/zephyrproject-rtos/zephyr/blob/966122045b9f555373db6b6eeed53cc0a84d76ea/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_iso.c#L117-L123